tests/test_closing.py

```diff
@@ -451,26 +435,91 @@ def get_fee_from_status(node, peer_id, i):
     wait_for(lambda: get_fee_from_status(funder, fundee_id, 0))
     wait_for(lambda: get_fee_from_status(fundee, funder_id, 1))
 
-    assert fee_mempool == fees_from_status[0]
-    assert fee_mempool == fees_from_status[1]
-    assert fee_mempool == opts['expected_close_fee']
+    assert opts['expected_close_fee'] == fees_from_status[0]
+    assert opts['expected_close_fee'] == fees_from_status[1]
+
+    # Get the closing transaction from the bitcoind mempool and get its fee
+
+    mempool = None
+    mempool_tx_ids = None
+
+    def get_mempool_when_size_1():
+        nonlocal mempool, mempool_tx_ids
+        mempool = bitcoind.rpc.getrawmempool(True)
+        mempool_tx_ids = list(mempool.keys())
+        return len(mempool_tx_ids) == 1
+
+    wait_for(get_mempool_when_size_1)
+
+    close_tx_id = mempool_tx_ids[0]
+    fee_mempool = round(mempool[close_tx_id]['fee'] * 10**8)
+
+    assert opts['expected_close_fee'] == fee_mempool
 
 
 def test_closing_fee(node_factory, bitcoind, chainparams):
-    """Test that the closing negotiation strategy works"""
+    """Test that the closing fee negotiation strategy works"""
     # feerate 27625 -> closing fee negotiation starts at 20000
     # feerate 29006 -> closing fee negotiation starts at 21000
 
     opts = {
         'funder_feerate_per_kw': 29006,
-        'fundee_feerate_per_kw': 27625,
-        'close_initiated_by': 'funder',
-        'expected_close_fee': 33533 if chainparams['elements'] else 20333
+        'fundee_feerate_per_kw': 27625
     }
 
+    opts['fee_negotiation_step'] = '50%'
+
+    opts['close_initiated_by'] = 'funder'
+    opts['expected_close_fee'] = 101 if chainparams['elements'] else 20334
```

Comment:
Here and below the fees for "elements" need to be checked and recorded. But how to run the test for that chain?

Reply:
You need to set `TEST_NETWORK=liquid-regtest` (and obv have the elements daemon available in your path).

Reply:
Thanks, updated.